kratos/includes/properties.h

```diff
@@ -97,10 +97,10 @@ class Properties : public IndexedObject
     ///@{
 
     /// Default constructor.
-    Properties(IndexType NewId = 0) : BaseType(NewId), mData(), mTables() {}
+    Properties(IndexType NewId = 0) : BaseType(NewId), mData(), mTables(), mpAccessor(this) {}
```

Comment:
I am not sure of the initialization you are making for the accessor.

here you initialize it with this (which is ok), but in the copy constructor i think it shouldalso be initialized with this, since it should act on the copied object.
I appreciate however that you want to deepcopy its configuration, not sure of how that should be accomplished

Reply:
If the accessor shall be retrieved through the properties then we need some default accessor
Passing `this` would only work if accessor derives from property